client/src/components/Forms/signup/index.jsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-console */
+import React, { useState } from 'react';
+import axios from 'axios';
+import {
+  Modal,
+  Button,
+  Form,
+  Input,
+  message,
+} from 'antd';
+import logo from '../../../assets/images/logo.png';
+import './index.css';
+
+const { Item } = Form;
+const { Password } = Input;
+
+function Signup() {
+  // eslint-disable-next-line no-unused-vars
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [fullName, setFullName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirm, setConfirm] = useState('');
+
+  const signUp = async () => {
+    try {
+      await axios.post('/api/signup', {
+        fullName,
+        email,
+        password,
+      });
+      setIsModalVisible(false);
+      message.success('Welcome to Nova');
+    } catch (err) {
+      message.error(`Sorry, try again , ${err.message}`);
+    }
+  };
+  const showModal = () => {
+    setIsModalVisible(true);
+  };
+  const handleCancel = () => {
+    setIsModalVisible(false);
+  };
+
+  const confirmPasswordHandel = (getFieldValue) => ({
+    validator(_, value) {
+      if (!value || getFieldValue('Password') === value) {
+        return Promise.resolve();
+      }
+      return Promise.reject(new Error('The two passwords that you entered do not match!'));
+    },
+  });
+
+  return (
+    <>
+      <Button type="primary" onClick={showModal}>
+        Open Modal
+      </Button>
+      <Modal
+        title={(
+          <div className="modal-logo-img">
```

Comment:
You need to split these into two components, one for the btn with the shown logic and one for the modal itself with the signup logic.